logic/strategy/card_utils.go

```diff
@@ -36,41 +36,28 @@ func otherOptions(desired int, avoid map[model.Card]struct{}) [][]model.Card {
 	return options
 }
 
-func chooseFrom(desired int, hand []model.Card) [][]model.Card { //nolint:gocyclo
-	if desired > len(hand) || desired > 4 || desired <= 0 {
-		return nil
-	}
-	// 6 choose 2 = 15, our largest number of potential deposits to the crib
-	allDeposits := make([][]model.Card, 0, 15)
-	for i, c1 := range hand {
-		if desired == 1 {
-			// it's a three or four player game, return one card
-			allDeposits = append(allDeposits, []model.Card{c1})
-			continue
+func chooseFrom(k int, hand []model.Card) [][]model.Card {
+	if k == 1 {
+		all := make([][]model.Card, len(hand))
+		for i, e := range hand {
+			all[i] = []model.Card{e}
 		}
-		for j := i + 1; j < len(hand); j++ {
-			c2 := hand[j]
-			if desired == 2 {
-				allDeposits = append(allDeposits, []model.Card{c1, c2})
-				continue
-			}
-			for k := i + 1; k < len(hand); k++ {
-				c3 := hand[k]
-				if desired == 3 {
-					allDeposits = append(allDeposits, []model.Card{c1, c2, c3})
-					continue
-				}
-				for l := i + 1; l < len(hand); l++ {
-					c4 := hand[l]
-					if desired == 4 {
-						allDeposits = append(allDeposits, []model.Card{c1, c2, c3, c4})
-						continue
-					}
-				}
-			}
+		return all
+	}
+	if k == len(hand) {
+		return [][]model.Card{hand}
+	}
+	all := make([][]model.Card, 0)
```

Comment:
`make`ing with 0 len isn't _super_ helpful. Could we `make` with cap 15, since we know most of the time we're not going to have more than that?

```go
all := make([][]model.Card, 0, 15)
```

what that does is reserve space for 15 elements, while filling in 0 of them.

Reply:
Max is theoretically 20 (6c3), but I don't think we ever actually choose 3? Anyway, we could also calculate the binomial coefficient and allocate the exact amount of space for each - which is better?

Reply:
I'd say it's better "just guess". 15 is fine, then golang will expand the slice as needed intelligently under the hood.

Reply:
My feeling is the extra work to calculate the binomial coefficient isn't worth it when we could just allocate a little extra memory, so I'll do this:

`all := make([][]model.Card, 0, 20)`